### PR TITLE
Update gems with security notices

### PR DESCRIPTION
**Pull request by robbkidd**
Gem updates to address security issues
* `shoulda-matchers` (to gem version, remove GitHub pinning since gem version now  includes the change in the commit to which we were pinned)
* `uglifier` had an issue (that doesn't affect current Supermarket, but upgrading because green is better than red on the vulnerability scanner)
* `chef` gem appears to be included to do version number validation, updated to a version that does not have conflicting RSpec version constraints. Upgrading RSpec is something that can be done later.

**UPDATED**

* `sidekiq` upgraded to version with fixes for vulnerabilities in the web UI (that supermarket does not use out-of-the-box)

### CODE REVIEW

**Comment by robbkidd** [comment]
Security updates for #1049 . That leaves Sidekiq security issues:

```
» be bundle-audit
Name: sidekiq
Version: 3.2.3
Advisory: 125675
Criticality: Unknown
URL: https://github.com/mperham/sidekiq/pull/2422
Title: Sidekiq Gem for Ruby Multiple Unspecified CSRF
Solution: upgrade to >= 3.4.2

Name: sidekiq
Version: 3.2.3
Advisory: 125676
Criticality: Unknown
URL: https://github.com/mperham/sidekiq/issues/2330
Title: Sidekiq Gem for Ruby web/views/queue.erb CurrentMessagesInQueue Element
Reflected XSS
Solution: upgrade to >= 3.4.0

Name: sidekiq
Version: 3.2.3
Advisory: 125678
Criticality: Unknown
URL: https://github.com/mperham/sidekiq/pull/2309
Title: Sidekiq Gem for Ruby web/views/queue.erb msg.display_class Element XSS
Solution: upgrade to >= 3.4.0
```

These don't affect Supermarket out of the box because we don't run the Sidekiq web interface.

**Comment by robbkidd** [comment]
```
» be bundle-audit update
Updating ruby-advisory-db ...
From https://github.com/rubysec/ruby-advisory-db
 * branch            master     -> FETCH_HEAD
Already up-to-date.
ruby-advisory-db: 233 advisories

» be bundle-audit
No vulnerabilities found
```

File: Gemfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ gem 'foreman'
 gem 'pundit'
 gem 'dotenv'
 gem 'octokit', require: false
-gem 'sidekiq'
+gem 'sidekiq', '= 3.4.2' # pinned to version prior to celluloid upgrade that brakes sidetiq
```
Comment (tas50) [inline Gemfile]: breaks

**Comment by smith** [comment]
:+1: 

Not sure on the preference of the rest of the team, but I find it much easier to work off of a branch of the repo than a fork, since if I want to check it out on my machine I have to add a remote from your fork and check that branch out, rather than just checking out a branch.

**Comment by robbkidd** [comment]
@smith Thanks! Merging.

On the process tangent, Nell and I have been working in forks for two reasons:

1. it's the way community contributors work without a commit bit
2. a fork is an additional guard against unintended pushes to branches in main repo

To your point, though, I agree there is an advantage to working in branches in the main repo: they're easier to get to—both for humans and for automatons. Reason No 2 above is probably mitigated by the git v2.0+ `config push.default` being `simple` instead of `matching` these days.

**Pull request merged**